### PR TITLE
feat: filter proposals

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary

This PR includes a filter popup to filter all proposals for various fields

## Changelog

```
- add filter component to porposal list
- type fitler
- participated flag to proposal
- participated filter
- status filter 
```

## Relevant Issues

Clsoes #4870 

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/locales/en.json [hunk 1]
```diff
@@ -1571,6 +1571,23 @@
...
+            "label": "Type",
+            "official": "Official",
+            "custom": "Custom"
+        },
+        "phase": {
+            "label": "Phase",
+            "upcoming": "Announcement",
+            "commencing": "Voting open",
+            "holding": "Counting starts",
+            "ended": "Counting stops"
```
Comment (Tuditi) [inline packages/shared/locales/en.json:1585]: Is it possible to use the same locale's as we use for the labels?

File: packages/shared/components/filters/Filter.svelte [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+    import type { ActivityFilter, AssetFilter } from '@core/wallet/interfaces'
     import { deepCopy } from '@core/utils'
     import type { Writable } from 'svelte/store'
+    import { ProposalFilter } from '@contexts/governance'
 
-    export let filterStore: Writable<Filter>
+    export let filterStore: Writable<ActivityFilter | AssetFilter | ProposalFilter>
 
-    let filter: Filter = deepCopy($filterStore)
+    let filter: ActivityFilter | AssetFilter | ProposalFilter = deepCopy($filterStore)
```
Comment (Tuditi) [inline packages/shared/components/filters/Filter.svelte:10]: What happened to Filter interface and why not keep it as a type union?

File: packages/shared/lib/contexts/governance/utils/isVotingForProposal.ts [hunk 3]
```diff
@@ -1,22 +1,13 @@
...
 export async function isVotingForProposal(
     proposalId: string,
     accountIndex = get(selectedAccountIndex)
 ): Promise<boolean> {
-    let overview: ParticipationOverview
-    if (accountIndex === get(selectedAccountIndex)) {
-        overview = get(participationOverview)
-    } else {
-        overview = await getParticipationOverview(accountIndex)
-    }
```
Comment (Tuditi) [inline packages/shared/lib/contexts/governance/utils/isVotingForProposal.ts:17]: Why did you remove this? It was an optimization to rely on the `participationOverview`, since that's a slow wallet API call.

**Review by Tuditi — CHANGES_REQUESTED** [review]
Just left some minor comments. Works well, good job!

File: packages/shared/lib/contexts/governance/utils/isVotingForProposal.ts (hunk 3 above)
Comment (MarkNerdi) [reply packages/shared/lib/contexts/governance/utils/isVotingForProposal.ts:17]: i moved it into another function `getParticipationsForProposal.ts` because i needed that part somewhere else

File: packages/shared/components/filters/Filter.svelte (hunk 2 above)
Comment (MarkNerdi) [reply packages/shared/components/filters/Filter.svelte:10]: i moved all of the general filter stuff into the `utils` module.
So now e.g. the `ActivityFilter` imports things from `utils`.
If i introduce again a generic `Filter` type, i'd need again to import eg. `ActivityFilter`, which already imports `utils`. 

I want to avoid this circular dependency, but i could create a `Filter`-type here in this component

File: packages/shared/locales/en.json (hunk 1 above)
Comment (MarkNerdi) [reply packages/shared/locales/en.json:1585]: wrt translations for filters, the structure isnt that flexible...

File: packages/shared/lib/contexts/governance/stores/proposal-filter.store.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { BooleanFilterOption } from '@core/utils/enums/filters'
+import { writable, Writable } from 'svelte/store'
+import { ProposalFilter, ProposalStatus, ProposalType } from '..'
+
+export const proposalFilter: Writable<ProposalFilter> = writable({
+    phase: {
+        active: false,
+        type: 'selection',
+        localeKey: 'filters.phase',
```
Comment (Tuditi) [inline packages/shared/lib/contexts/governance/stores/proposal-filter.store.ts:9]: Can we update this then to alleviate the burden on the translaters and to keep sure it's consistent across all languages? I guess the only thing that would need add is a label, right?

```suggestion
        localeKey: 'proposalStatus',
```

**Review by Tuditi — CHANGES_REQUESTED** [review]

File: packages/shared/components/filters/FilterItem.svelte [hunk 5]
```diff
@@ -24,7 +24,7 @@
 <div class="filter-item border-t border-solid border-gray-200 dark:border-gray-800 ">
     <div class="px-4 py-2 flex flex-row justify-between">
         <Checkbox
-            label={localize(filterUnit.localeKey + '.label')}
+            label={localize(filterUnit.labelKey)}
```
Comment (Tuditi) [inline packages/shared/components/filters/FilterItem.svelte:27]: Maybe labelKey can be made optional and default to the one you previously had to remove all the duplication?

**Review by Tuditi — APPROVED** [review]
LGTM! :fire:
